test/langtools/tools/javac/lambda/ClassInIntersectionTypeTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * @test /nodynamiccopyright/
+ * @bug 8322810
+ * @summary Lambda expressions can implement classes
+ * @compile/fail/ref=ClassInIntersectionTypeTest.out -XDrawDiagnostics ClassInIntersectionTypeTest.java
+ */
+
+import java.io.Serializable;
+import java.lang.annotation.Annotation;
+
+public class ClassInIntersectionTypeTest {
+    // test 1
+    void m1() {
+        ClassInIntersectionTypeTest r1 = (ClassInIntersectionTypeTest & Runnable) () -> System.out.println("Hello, World!");
+        ClassInIntersectionTypeTest r2 = (ClassInIntersectionTypeTest & Runnable) ClassInIntersectionTypeTest::run1;
+    }
+
+    static void run1() {}
+
+    // test 2
+    static void foo() {
+        run2(() -> System.out.println("Hello, World!"));
+        run2(ClassInIntersectionTypeTest::run1);
+    }
+
+    static <T extends ClassInIntersectionTypeTest & Runnable> void run2(T t) {
+        t.run();
+    }
+
+    static Class<? extends Annotation> myAnnoType() { return null; }
+    @interface Anno {}
+    Anno a = (Anno & Serializable) ()-> null; // annotations not allowed
```

Comment:
Actually, I wonder -- the annotation should not be allowed in any position right? Not just in the first type? E.g. since an annotation is an interface, you can also have:

`Runnable & Anno1 & Anno2`

Do we detect those?

Reply:
I will add a test

Reply:
done